pkg/cvo/sync_worker.go

```diff
@@ -530,10 +535,34 @@ type consistentReporter struct {
 	reporter  StatusReporter
 }
 
-func (r *consistentReporter) Inc() {
+// Begin task appends the given task to status.Current.
+func (r *consistentReporter) BeginTask(task *payload.Task) {
+	r.lock.Lock()
+	defer r.lock.Unlock()
+	r.status.Current = append(r.status.Current, task)
+}
+
+// FinishTask atomically removes the given task from status.Current
+// and, if success is true, increments done.
+func (r *consistentReporter) FinishTask(task *payload.Task, success bool) {
 	r.lock.Lock()
 	defer r.lock.Unlock()
-	r.done++
+
+	if success {
+		r.done++
+	}
+
+	for i, tsk := range r.status.Current {
+		if tsk == task {
+			copy(r.status.Current[i:], r.status.Current[i+1:])
```

Comment:
This is not great :(

Reply:
> This is not great :(

This is the recommended way to delete from a slice of pointers while preserving ordering, per [this][1].  Did you have an alternative you prefer?

[1]: https://github.com/golang/go/wiki/SliceTricks